trade-module/src/main/java/io/edpn/backend/trade/domain/controller/v1/TradeModuleController.java

```diff
@@ -1,22 +1,34 @@
 package io.edpn.backend.trade.domain.controller.v1;
 
 import io.edpn.backend.trade.application.dto.v1.CommodityMarketInfoResponse;
+import io.edpn.backend.trade.application.dto.v1.FindCommodityRequest;
+import io.edpn.backend.trade.application.dto.v1.FindCommodityResponse;
 import io.edpn.backend.trade.application.dto.v1.LocateCommodityRequest;
 import io.edpn.backend.trade.application.dto.v1.LocateCommodityResponse;
 import jakarta.validation.Valid;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @RequestMapping("/api/v1/trade")
 public interface TradeModuleController {
 
     @GetMapping("/best-price")
     List<CommodityMarketInfoResponse> getBestCommodityPrice();
 
-    @PostMapping("/locate-commodity")
-    List<LocateCommodityResponse> locateCommodityWithFilters(@Valid @RequestBody LocateCommodityRequest locateCommodityRequest);
+    @GetMapping("/locate-commodity/filter")
+    List<LocateCommodityResponse> locateCommodityWithFilters(@Valid LocateCommodityRequest locateCommodityRequest);
+    
+    @GetMapping("/commodity/{id}")
+    Optional<FindCommodityResponse> findValidatedCommodityById(@PathVariable UUID id);
```

Comment:
I would prefer a natural key to be used instead of the internal module specific UUID